plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -823,34 +822,116 @@ protected function get_and_maybe_update_months_filter_cache(): array {
+					$cached_value[0]->month !== $current_year_month->month )
 			) {
 				array_unshift( $cached_value, $current_year_month );
 			}
 			return $cached_value;
 		}
 
+		// Option to use a simplified, more performant method for building the months options list.
+		$use_simplified_months_options = boolval( get_option( 'wc_' . $this->order_type . '_list_table_months_filter_simplified', false ) );
+
+		if ( true === $use_simplified_months_options ) {
+			$order_dates = $this->get_simplified_months_filter_options();
+		} else {
+			$orders_table = esc_sql( OrdersTableDataStore::get_orders_table_name() );
+			$trash_status = esc_sql( OrderStatus::TRASH );
+			$utc_offset   = wc_timezone_offset();
+
+			$order_dates = $wpdb->get_results(
+				// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				$wpdb->prepare(
+					"
+						SELECT DISTINCT YEAR( t.date_created_local ) AS year,
+										MONTH( t.date_created_local ) AS month
+						FROM ( SELECT DATE_ADD( date_created_gmt, INTERVAL $utc_offset SECOND ) AS date_created_local FROM $orders_table WHERE type = %s AND status != '$trash_status' ) t
+						ORDER BY year DESC, month DESC
+					",
+					$this->order_type
+				)
+				// phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			);
+		}
+
+		update_option( $cache_option_date_name, time(), false );
+		update_option( $cache_option_value_name, $order_dates, false );
+
+		return $order_dates;
+	}
+
+	/**
+	 * Get a list of year-month options for filtering the orders list table.
+	 *
+	 * This is a simplified approach that finds the oldest order and generates a year-month option for every month
+	 * in the range between then and the current month. It uses a query that is much more performant on sites with a
+	 * large number of orders. The tradeoff is that it might include months for which no orders actually exist.
+	 *
+	 * @return \stdClass[] List of year-months.
+	 */
+	protected function get_simplified_months_filter_options(): array {
+		global $wpdb;
+
 		$orders_table = esc_sql( OrdersTableDataStore::get_orders_table_name() );
-		$utc_offset   = wc_timezone_offset();
+		$trash_status = esc_sql( OrderStatus::TRASH );
 
-		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
-		$order_dates = $wpdb->get_results(
+		$first_year_month = $wpdb->get_row(
+			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
 			$wpdb->prepare(
 				"
-					SELECT DISTINCT YEAR( t.date_created_local ) AS year,
-									MONTH( t.date_created_local ) AS month
-					FROM ( SELECT DATE_ADD( date_created_gmt, INTERVAL $utc_offset SECOND ) AS date_created_local FROM $orders_table WHERE type = %s AND status != 'trash' ) t
-					ORDER BY year DESC, month DESC
+					SELECT YEAR( t.date_created_gmt ) AS year,
+					       MONTH( t.date_created_gmt ) AS month
+					FROM $orders_table t
+					WHERE type = %s
+					AND status != '$trash_status'
+					ORDER BY year ASC, month ASC
```

Comment:
Just curious if we've tested impact of `ORDER BY` on this query.  I'm not sure that we have alternatives (other than using ID and hoping 🤞  things are chronological), but this clause tends to be one that tends to come up a lot on larger sites.

Reply:
Haven't tested it, but I don't think we can rely on the ID being chronological. If a store has ever imported orders from another site or a 3rd party system, they could be from any date but they would still increment the ID column as usual.